packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -308,15 +308,8 @@ Future<void> buildGradleApp({
   if (target != null) {
     command.add('-Ptarget=$target');
   }
-  assert(buildInfo.trackWidgetCreation != null);
-  command.add('-Ptrack-widget-creation=${buildInfo.trackWidgetCreation}');
+  command.addAll(androidBuildInfo.buildInfo.toGradleConfig());
 
-  if (buildInfo.extraFrontEndOptions != null) {
```

Comment:
To avoid future refactor breakage, I moved the logic for encoding the various build options into the build config, similar to how it is done for desktop. This changed a few of the expectations.